episodes/25-linreg-diagnostics.Rmd

```diff
@@ -0,0 +1,168 @@
+* No one residual is visibly away from the random pattern of the residuals indicating that there are no outliers.
+
+```{r}
+ols_plot_resid_fit(model1)
+```
+
+### Residual Histogram
+Additionally, we can create a histogram of residuals for detecting violation of normality assumption.
+
+```{r}
+ols_plot_resid_hist(model1)
+```
+
+## Additional Diagnostics
+In addition the residual diagnostics, we can also assess our model for Heteroskedasticity, Multicollinearity and any Influential/High leverage points.
+
+### Heteroskedasticity Check
+
+We can use the ncvTest() function to test for equal variances.
+
+```{r}
+ncvTest(model1)
+```
+The p-value here is low, indicating that this model may have a problem of unequal variances.
+
+### Multicollinearity Check
+We want to know whether we have too many variables that have high correlation with each other.
+If the value of GVIF is greater than 4, it suggests collinearity.
+
+```{r}
+vif(model1)
+```
+
+### Outlier Identification
+We can identify outliers in our data by creating a QQPlot and running statistical tests on the data.
+The outlierTest() reports the Bonferroni p-values for testing each observation in turn to be a mean-shift outlier, based Studentized residuals in
+linear (t-tests), generalized linear models (normal tests), and linear mixed models.
+
+```{r}
+# Create a qqPlot
+qqPlot(model1, id.n = 2)
+
+# Test for outliers
+outlierTest(model1)
+```
+
+The null hypothesis for the Bonferonni adjusted outlier test is the observation is an outlier. Here observation related to ‘4612’ is an outlier.
+In our data '4612' is difficult to identify as it may refer to either a Living Environment or a Barriers rank.
+
+### Identifying Influence Points
+
+We can identify potential influential points via an Influence Plot, these may be Ouliers and/or High Leverage points.
+A point can be considered influential if it's exclusion causes substantial change in the estimated regression.
+
+An influence plot summarises the 3 metrics for influence analysis in one single glance.
+The X-axis plots the leverage, which is normalised between 0 and 1.
+The Y-axis plots the studentized residuals, which can be positive or negative.
+The size of the circles for each data point reflect its Cook’s distance, degree of influence.
+
+Vertical reference lines are drawn at twice and three times the average hat value, horizontal reference lines at -2, 0, and 2 on the Studentized-residual scale.
+```{r}
+# Influence Plots
+influencePlot(model1)
+```
+Identified influential points are returned in a data frame with the hat values, Studentized residuals and Cook's distance of the identified points.
+If no points are identified, nothing is returned.
+
+Influence points can be further explored with an Influence Index Plot which provides index plots of influence and related diagnostics for a regression model.
+
+```{r}
+influenceIndexPlot(model1, id.n = 3)
```

Comment:
A _lot_ of warnings here:

```
Warning in plot.window(...): "id.n" is not a graphical parameter
Warning in plot.xy(xy, type, ...): "id.n" is not a graphical parameter
Warning in axis(side = side, at = at, labels = labels, ...): "id.n" is not a
graphical parameter
Warning in axis(side = side, at = at, labels = labels, ...): "id.n" is not a
graphical parameter
Warning in box(...): "id.n" is not a graphical parameter
Warning in title(...): "id.n" is not a graphical parameter
... # truncated output
```

Looks like they're all coming from `id.n` not being a valid argument?

Reply:
id.n = 3 removed, plot builds ok without it, no errors